src/main/java/org/alliancegenome/curation_api/services/validation/dto/fms/HTPExpressionDatasetSampleAnnotationFmsDTOValidator.java

```diff
@@ -0,0 +1,257 @@
+import org.alliancegenome.curation_api.model.entities.Note;
+import org.alliancegenome.curation_api.model.entities.TemporalContext;
+import org.alliancegenome.curation_api.model.entities.VocabularyTerm;
+import org.alliancegenome.curation_api.model.entities.ontology.MMOTerm;
+import org.alliancegenome.curation_api.model.entities.ontology.NCBITaxonTerm;
+import org.alliancegenome.curation_api.model.entities.ontology.OBITerm;
+import org.alliancegenome.curation_api.model.ingest.dto.fms.HTPExpressionDatasetSampleAnnotationFmsDTO;
+import org.alliancegenome.curation_api.model.ingest.dto.fms.WhereExpressedDTO;
+import org.alliancegenome.curation_api.response.ObjectResponse;
+import org.alliancegenome.curation_api.response.SearchResponse;
+import org.alliancegenome.curation_api.services.AffectedGenomicModelService;
+import org.alliancegenome.curation_api.services.AlleleService;
+import org.alliancegenome.curation_api.services.DataProviderService;
+import org.alliancegenome.curation_api.services.ExternalDataBaseEntityService;
+import org.alliancegenome.curation_api.services.VocabularyTermService;
+import org.alliancegenome.curation_api.services.ontology.MmoTermService;
+import org.alliancegenome.curation_api.services.ontology.NcbiTaxonTermService;
+import org.alliancegenome.curation_api.services.ontology.ObiTermService;
+import org.alliancegenome.curation_api.services.ontology.StageTermService;
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.lang.StringUtils;
+
+import jakarta.enterprise.context.RequestScoped;
+import jakarta.inject.Inject;
+import jakarta.transaction.Transactional;
+
+@RequestScoped
+public class HTPExpressionDatasetSampleAnnotationFmsDTOValidator {
+
+	@Inject ExternalDataBaseEntityFmsDTOValidator externalDataBaseEntityFmsDtoValidator;
+	@Inject GeneExpressionAnnotationFmsDTOValidator geneExpressionAnnotationFmsDTOValidator;
+	@Inject HTPExpressionDatasetSampleAnnotationDAO htpExpressionDatasetSampleAnnotationDAO;
+	@Inject VocabularyTermService vocabularyTermService;
+	@Inject ExternalDataBaseEntityService externalDataBaseEntityService;
+	@Inject DataProviderService dataProviderService;
+	@Inject ObiTermService obiTermService;
+	@Inject MmoTermService mmoTermService;
+	@Inject StageTermService stageTermService;
+	@Inject AlleleService alleleService;
+	@Inject AffectedGenomicModelService affectedGenomicModelService;
+	@Inject NcbiTaxonTermService ncbiTaxonTermService;
+
+	@Transactional
+	public HTPExpressionDatasetSampleAnnotation validateHTPExpressionDatasetSampleAnnotationFmsDTO(HTPExpressionDatasetSampleAnnotationFmsDTO dto, BackendBulkDataProvider backendBulkDataProvider) throws ValidationException {
+		ObjectResponse<HTPExpressionDatasetSampleAnnotation> htpSampleAnnotationResponse = new ObjectResponse<>();
+		HTPExpressionDatasetSampleAnnotation htpSampleAnnotation;
+
+		if (dto.getSampleId().getPrimaryId() != null) {
```

Comment:
I prefer to use StringUtils.isNotBlank for checks on strings as it rules out empty or whitespace-only strings as well as nulls.